src/CommunityToolkit.Maui.UnitTests/Mocks/AnimationExtensions.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Maui;
+using Microsoft.Maui.Animations;
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Handlers;
+
+namespace CommunityToolkit.Maui.UnitTests.Mocks;
+
+static class AnimationExtensions
+{
+	public static void EnableAnimations(this IView view) => MockAnimationHandler.Prepare(view);
+
+	// Inspired by Microsoft.Maui.Controls.Core.UnitTests.AnimationReadyHandler
+	class MockAnimationHandler : ViewHandler<IView, object>
+	{
+		MockAnimationHandler(IAnimationManager animationManager) : base(new PropertyMapper<IView>())
+		{
+			SetMauiContext(new AnimationEnabledMauiContext(animationManager));
+		}
+
+		MockAnimationHandler() : this(new TestAnimationManager(new AsyncTicker()))
+		{
+		}
+
+		public IAnimationManager? AnimationManager => ((AnimationEnabledMauiContext?)MauiContext)?.AnimationManager;
+
+		public static T Prepare<T>(T view) where T : IView
+		{
+			view.Handler = new MockAnimationHandler();
+
+			return view;
+		}
+
+		protected override object CreateNativeView() => new();
+
+		class AnimationEnabledMauiContext : IMauiContext, IServiceProvider
+		{
+			public AnimationEnabledMauiContext(IAnimationManager manager)
+			{
+				AnimationManager = manager;
+			}
+
+			public IServiceProvider Services => this;
+
+			public IMauiHandlersServiceProvider Handlers => throw new NotImplementedException();
+
+			public IAnimationManager AnimationManager { get; }
+
+			public object GetService(Type serviceType)
+			{
+				if (serviceType == typeof(IAnimationManager))
+					return AnimationManager;
+
+				throw new NotSupportedException();
+			}
+		}
+
+		class AsyncTicker : Ticker
+		{
+			CancellationTokenSource? _cancellationTokenSource;
+
+			public override async void Start()
+			{
+				_cancellationTokenSource = new();
+
+				while (!_cancellationTokenSource.IsCancellationRequested)
+				{
+					Fire?.Invoke();
+
+					if (!_cancellationTokenSource.IsCancellationRequested)
+						await Task.Delay(16);
```

Comment:
I assume you have actively chosen not to pass the `_cancellationTokenSource.Token` in to `Task.Delay` to avoid exceptions and given 16ms really is a short delay

Reply:
That said would a `TimeSpan.FromMilliseconds` make this slightly more readable?

Reply:
> chosen not to pass the _cancellationTokenSource.Token in to Task.Delay to avoid exceptions
Yup! 💯 

> would a TimeSpan.FromMilliseconds make this slightly more readable

Also yup 👍